01_fib/alfredxiao/main.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func fibSeries(n int) []int {
+	if n >= 93 || n <= -93 {
+		panic("Numbers beyond 93 or -93 are not supported!")
+	}
+
+	sign := 1
+	if n < 0 {
+		sign = -1
+	}
+
+	series := make([]int, n*sign+1)
+	if n == 0 {
+		series[0] = 0
+	} else {
+		series[0], series[1] = 0, 1
```

Comment:
Since you do `series[0] = 0` in some form in both branches, you can move that outside the conditional. You can then early return for the `n==0` case, and move the main code back one level to the left. See https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88 for more in-depth reasoning. Idiomatic Go tends to use this style.

Reply:
Or,
```
series[0] = 0
if n != 0 {
  series[1] = 1
}
for i := 2; i <= n*sign; i++ {
  ...
}
return series
```
This avoids the early return looking like it might be an error case (which an early return most often is).